### ISSUE TITLE
Only depend on required d3 packages

### ISSUE THREAD

**Issue by jmalonzo** [PMSI-AlignAlytics/dimple#209]
There are now different packages for D3 (d3-selection, d3-ease, etc..). I think it would be nice if we depend on those packages instead of the full d3 package.

Thanks

**Comment by IPWright83** [comment]
These modules are for the new D3 version 4 and aren't officially released yet, but does seem like a good idea.

**Comment by curran** [comment]
I would agree. The modules are *released* on their own, independently from the top-level D3 package (e.g. see [d3-selection releases](https://github.com/d3/d3-selection/releases)), but D3 4.0 is not released yet. I think it's a pretty safe bet to migrate to the new packages.

I opened a similar issue for Scrollgrid https://github.com/PMSI-AlignAlytics/scrollgrid/issues/7
